auth-api/src/auth_api/services/authorization.py

```diff
@@ -109,18 +109,27 @@ def get_user_authorizations_for_entity(business_identifier: str, expanded: bool
                     permissions = PermissionsService.get_permissions_for_membership(auth.status_code, 'SYSTEM')
                     auth_response['roles'] = permissions
         else:
-            keycloak_guid = user_from_context.sub
-            if business_identifier and keycloak_guid:
-                auth = AuthorizationView.find_user_authorization_by_business_number(
-                    business_identifier=business_identifier,
-                    keycloak_guid=keycloak_guid,
-                    org_id=user_from_context.account_id
-                )
+            if business_identifier and (org_id := user_from_context.account_id):
 
-            if auth:
-                permissions = PermissionsService.get_permissions_for_membership(auth.status_code, auth.org_membership)
-                auth_response = Authorization(auth).as_dict(expanded)
-                auth_response['roles'] = permissions
+                # if this is an API GW account, check if the account has access to the resource
+                if user_from_context.login_source == LoginSource.API_GW.value:
```

Comment:
I think this should happen in legal-api, not auth-api.. the function you're changing is `get_user_authorizations_for_entity`, when there is already a function for `get_account_authorizations_for_org` on line 72